### PR TITLE
Ts intro

### PR DESCRIPTION
**Pull request by hyanwong**
Here's a first pass at the start of an intro section. I've only really worked up the detail under the section "A record of full genetic ancestry": the two other sections aren't worked up.

More generally, I wonder if we should hide the detailed stuff from the casual site visitor? That's why I've put the "three key things a ts can provide" as single sentences at the top. I think of these as a TLDR summary. Each has a "Read more..." link, which currently jumps to further down the page, but I imagine could jump to another page, or could open an extra hidden box (like the html5 `summary` tag), or something. That would mean that all we have visible to the casual viewer, above the Tutorials section, is this:

> ### What is a tree-sequence
> A succinct tree sequence, or tree sequence for short, represents the relationships between a set of DNA sequences. Tree sequences can be used to store genetic data efficiently, and enable powerful and rapid analysis of millions of whole genomes at a time. They can be created by simulation or by inferring relationships from genetic variation.
> 
> The key things a tree sequence can provide are:
> * **A record of full genetic ancestry**: Tree sequences efficiently store the full genetic history of a set of genomes, by sharing edges (branches) between adjacent trees. Read more ...
> * **An encoding of DNA data**: Placing mutations on a tree sequence allows lossless representation and compression of DNA datasets. Read more ...
> * **An efficient analysis framework**: Analyses of genetic variation can be reformulated as measurements on trees; applying this to trees with shared structure can (massively?) speed up genome-wide statistical calculations. Read more ...

I wondered if we wanted to add animations to these 3 points too (I have possibly suitable ones for the first two). But maybe that would be too distracting?

### CODE REVIEW

**Comment by jeromekelleher** [comment]
I don't know about animations, but the text is great.

**Comment by hyanwong** [comment]
> I don't know about animations, but the text is great.

Thanks @jeromekelleher ! Any comments about the "read more" text / section? Is this the right sort of depth? Would it be too long to put on the main "learn" page? Here it is for reference:

> ### A record of full genetic ancestry
> 
> Two basic principles underpin the idea of tree sequences:
> 
> * At a given position along a chromosome, the relationship between a set of genomes can be represented by a single evolutionary tree.
> * Trees near to each other on the chromosome look similar, that is they share structure.
> 
> A visualization of a short tree sequence is shown below, focussing on the first 1000 letters (base pairs) of 20 simulated genomes. These genomes are represented as blue dots at the tips of each tree; the position of the trees along the chromosome is indicated underneath.
> Screenshot 2021-01-21 at 17 11 46
> 
> [ A tree sequence of the first 1000 base pairs of a simulated chromosome. Branches, or edges, shared between trees are highlighted ]
> 
> The relationship, or genealogy, linking the 20 genomes is the same between positions 0 and 383, and is shown by the first tree. This is slightly different from the genealogy between positions 384 and 874 (the second tree), which is again slightly different from the genealogy between positions 875 and 999 (the third tree). Nevertheless the three trees are almost identical, as highlighted by the shared, coloured branches.
> A tree sequence captures the shared structure between trees by allowing edges (the branches or lines in the trees) to persist from one tree to another. This is equivalent to storing the difference between adjacent trees. Differences between adjacent trees are usually very small, even for huge trees, which explains why tree sequences are so efficient.

**Comment by petrelharp** [comment]
Can we add `Gemfile.lock` to the `.gitignore`?

**Comment by petrelharp** [comment]
Hm, I'm trying to preview this but I'm not getting any of the styling or most of the images. I did the steps in the README - any ideas what's going wrong?

File: learn.html [hunk 1]
```diff
@@ -7,12 +7,65 @@
...
+    <dl>
+        <dt>A record of full genetic ancestry</dt>
+        <dd>Tree sequences efficiently store the full genetic history of a set of genomes, by sharing
+        <em>edges</em> (branches) between adjacent trees. <a href="#ancestry">Read more ...</a></dd>
+        <dt>An encoding of DNA data</dt>
+        <dd>Placing mutations on a tree sequence allows lossless representation and compression of
+    DNA datasets. <a href="#mutations">Read more ...</a></dd>
+        <dt>An efficient analysis framework</dt>
+        <dd>Analyses of genetic variation can be reformulated as measurements on trees;
+    applying this to trees with shared structure can (massively?) speed up genome-wide statistical calculations. <a href="#analysis">Read more ...</a></dd>
```
Comment (petrelharp) [inline learn.html:27]: "can massively" I think is ok; if it feels too hyperbolic, we could say "often speeds up"

File: learn.html [hunk 2]
```diff
@@ -7,12 +7,65 @@
...
+    </dl>
+    
+    <h3 id="ancestry">A record of full genetic ancestry</h3>
+    <p>Two basic principles underpin the tree sequence concept:
+    <ol>
+        <li>At a given position along a chromosome, the relationship between a set of
+        genomes can be represented by a single evolutionary tree.</li>
```
Comment (petrelharp) [inline learn.html:34]: delete "evolutionary"

File: learn.html [hunk 3]
```diff
@@ -7,12 +7,65 @@
...
+    indicated underneath.
+    <figure>
+    {% include assets/images/basic_ts.svg %}
+      <figcaption>A tree sequence of the first 1000 base pairs of a
+      simulated chromosome. Branches, or <em>edges</em>, shared between trees are highlighted</figcaption>
+</figure>
+    The relationship, or genealogy, linking the 20 genomes is the same between positions
+    0 and 383, and is shown by the first tree. This is slightly different from the genealogy
+    between positions 384 and 874 (the second tree), which is again slightly different
+    from the genealogy between positions 875 and 999 (the third tree). Nevertheless the
```
Comment (petrelharp) [inline learn.html:50]: you could say "the last position" rather than "position 999" to avoid people getting distracted thinking about zero-based indexing

**Comment by petrelharp** [comment]
This is great! Very understandable and concise. The thing I'd want to add to this is something about how the trees give us strictly more information - like, we often want to know the trees! Right now, the explanation is focused on "how it works", but doesn't talk about how the tree sequence tells us about *actual ancestors* (or tries to if it's inferred). (This is as opposed to some arbitrary haplotype compression scheme, which much of what's written would apply to as well.)

File: learn.html (hunk 2 above)
Comment (hyanwong) [reply learn.html:34]: Why? I put it in there to made it completely clear that we aren't talking about a plant. Is "evolutionary" wrong? I could say "gene tree" but that's confusing to a molecular biologist.

**Comment by hyanwong** [comment]
> Can we add `Gemfile.lock` to the `.gitignore`?

I had a brief chat about this to @benjeffery and I think it's always recommended to include it. Can't you just update it with your settings?

**Comment by hyanwong** [comment]
> This is great! Very understandable and concise. 

Thanks @petrelharp . Did you sort out previewing it? I see that the instructions in https://github.com/tskit-dev/tskit-site/pull/12 haven't been merged yet, so maybe all you need is to do `bundle exec jekyll serve` to start the web server and go to http://127.0.0.1:4000

> The thing I'd want to add to this is something about how the trees give us strictly more information - like, we often want to know the trees! 

That's a good point. 

> Right now, the explanation is focused on "how it works", but doesn't talk about how the tree sequence tells us about _actual ancestors_ (or tries to if it's inferred). (This is as opposed to some arbitrary haplotype compression scheme, which much of what's written would apply to as well.)

Maybe a sentence identifing the MRCAs in the trees as actual ancestors would be a good lead-in to this. Are you thinking of putting this in the "Read more..." section? And on that note, do you think the extra detail with plots etc should be on another page or hidden somehow, or simply e.g. further down the main "learn.html" page?

File: learn.html (hunk 2 above)
Comment (petrelharp) [reply learn.html:34]: The trees don't have anything to do with evolution, do they?

**Comment by petrelharp** [comment]
> The thing I'd want to add to this is something about how the trees give us strictly more information - like, we often want to know the trees!

Well, my comment wasn't quite right, since the "record of full genetic ancestry" does talk about how it's recording genetic ancestry. But it could make the point a bit more directly.

> Maybe a sentence identifing the MRCAs in the trees as actual ancestors would be a good lead-in to this. Are you thinking of putting this in the "Read more..." section? And on that note, do you think the extra detail with plots etc should be on another page or hidden somehow, or simply e.g. further down the main "learn.html" page?

Hm, well I think that the point could be emphasized in the "record of full genetic ancestry" section (that you get to after clicking "read more..."). And I like it the way it is - just further down the page - rather than hiding the details somehow.  Well, depends what we want to do with this page, I guess: right now when you go to it you can't tell that the tutorials, talks, and publications are further down the page.  So, maybe this page is "a list of resources", in which case the detail with paragraphs and figures should actually be on a different page you click through to (or is hidden).

**Comment by jeromekelleher** [comment]
I think the way forward here to get a version of this merged and iterate from there - @hyanwong, are there any updates you would like to make or are you happy to merge?

**Comment by benjeffery** [comment]
@hyanwong I'm going to make a few edits here and merge, we can always iterate in a subsequent PR.

**Comment by hyanwong** [comment]
> @hyanwong I'm going to make a few edits here and merge, we can always iterate in a subsequent PR.

Hang on - let me push my recent changes first

**Comment by hyanwong** [comment]
Done. The " **Extra sentence from Peter?**" bit can be swapped for one of Peter's 2 suggestions:

1. useful in many ways THAT ARE ALL TOTALLY AWESOME.
2. useful in many ways, for instance, to identify the origin and age of alleles under selection, to describe geographic patterns of ancestry, or just to gain a more concrete understanding of the sources of signal and noise affecting the genetic data.

I quite like 1 :)

**Pull request merged**